C++/source/graph/Dijkstra.cpp

```diff
@@ -0,0 +1,49 @@
+// Author : Abdelrahman Hammam
+//Github handle : AbdoHammam
+#include<bits/stdc++.h>
+#define PB push_back
+#define MP make_pair
+typedef long long ll;
+using namespace std;
+vector<int> Dist; // Distance between the root and the other nodes
+vector<pair<int,int> > *G;// Graph nodes , adj list and cost
+void Dijkstra(int source, int N)
+{
+    priority_queue<pair<int,int>, vector<pair<int,int>>, greater<pair<int,int> > > Q; // Min heap
+    Dist.assign(N,1e9);
+    Dist[source] = 0;
+    Q.push({0,source});
+    while(!Q.empty())
+    {
+        int u = Q.top().second;
+        Q.pop();
+        for(auto &c : G[u])
+        {
+            int v = c.second;
+            int w = c.first;
+            if(Dist[v] > Dist[u]+w)
+            {
+                Dist[v] = Dist[u]+w;
+                Q.push({Dist[v],v});
+            }
+        }
+    }
+}
+int main()
```

Comment:
Please remove `main()` and convert this to a header file, that should be located at  `C++/include/algorithm/graph/djikstra.hpp`.